includes/sensei-functions.php

```diff
@@ -281,14 +281,48 @@ function sensei_user_login_url() {
  * duplicate of Sensei()->access_settings().
  *
  * @since 1.9.0
+ * @since 3.5.2 Added hook to filter the return.
  * @return bool
  */
 function sensei_is_login_required() {
+	global $post;
 
-	$login_required = isset( Sensei()->settings->settings['access_permission'] ) && ( true == Sensei()->settings->settings['access_permission'] );
+	$post_type = get_post_type( $post );
+	$course_id = null;
+
+	switch ( $post->post_type ) {
```

Comment:
I wonder if we need to do a check here for `$post` in case this gets called outside the loop. Alternatively, you could just use the unused `$post_type` from above. That should essentially do the same thing, I think. That should never have a value if `$post` isn't set.

Reply:
Oops! I messed the things up in a commit refactor, but I fixed the original commit here: https://github.com/Automattic/sensei/pull/3631/commits/09ba45e5dd935906d59b630d6cb9bd0420aa0663

I just didn't understand what you meant with `We could then use $post->ID below.`

Reply:
For the `$post->ID` bit, I meant even though we aren't explicitly checking `$post` is set, `$post_type` will should never have a true-like value without `$post` being set (so we should be okay here).